src/main/java/pl/edu/mimuw/Main.java

```diff
@@ -1,18 +1,34 @@
 package pl.edu.mimuw;
 
 import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.Iterator;
+import java.util.stream.Stream;
 
 public class Main {
+  private static Stream<BankAction> actionsGenerator() {
```

Comment:
fajowo 😎  ciekawsze są jednak operacje na streamach (modyfikacje jego elementów), więc warto zerknąć na metody takie jak `map` i `filter` oraz to jak tak przerobione Streamy zbierać funcją `collect(...)`